judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CastingFunctionsTest.java

```diff
@@ -0,0 +1,128 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import hu.blackbelt.judo.runtime.core.jsl.itest.castingfunctionsmodel.guice.castingfunctionsmodel.CastingFunctionsModelDaoModules;
+import hu.blackbelt.judo.runtime.core.jsl.itest.castingfunctionsmodel.sdk.castingfunctionsmodel.castingfunctionsmodel.*;
+import hu.blackbelt.judo.runtime.core.jsl.itest.castingfunctionsmodel.sdk.castingfunctionsmodel.castingfunctionsmodel.B.BDao;
+import hu.blackbelt.judo.runtime.core.jsl.itest.castingfunctionsmodel.sdk.castingfunctionsmodel.castingfunctionsmodel.CA.CADao;
+import hu.blackbelt.judo.runtime.core.jsl.itest.castingfunctionsmodel.sdk.castingfunctionsmodel.castingfunctionsmodel.Tester.TesterDao;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+public class CastingFunctionsTest extends AbstractJslTest {
+
+    @Inject BDao bDao;
+    @Inject CADao caDao;
+    @Inject TesterDao testerDao;
+
+    @Override
+    public Module getModelDaoModule() {
+        return new CastingFunctionsModelDaoModules();
+    }
+
+    @Override
+    public String getModelName() {
+        return "CastingFunctionsModel";
+    }
+
+    @Test
+    public void test() {
+        CA ca = caDao.create(CA.builder().withNameA("aca1").withNameB("bca1").withNameCA("ca1").build());
+        CA ca1 = caDao.create(CA.builder().withNameA("aca2").withNameB("bca2").withNameCA("ca2").build());
+        CA ca2 = caDao.create(CA.builder().withNameA("aca3").withNameB("bca3").withNameCA("ca3").build());
+        B caAsB = bDao.getById(ca.get__identifier()).orElseThrow();
+        B caAsB1 = bDao.getById(ca1.get__identifier()).orElseThrow();
+        B caAsB2 = bDao.getById(ca2.get__identifier()).orElseThrow();
+
+        Tester tester = testerDao.create(Tester.builder()
+                                               .withB(bDao.create(B.builder().withNameA("ab").withNameB("b").build()))
+                                               .withBs(List.of(
+                                                       bDao.create(B.builder().withNameA("ab1").withNameB("b1").build()),
+                                                       bDao.create(B.builder().withNameA("ab2").withNameB("b2").build())
+                                               ))
+                                               .withCaAsB(caAsB)
+                                               .withCaAsBs(List.of(caAsB1, caAsB2))
+                                               .build());
+
+        // kind of
```

Comment:
We should split this test into as many parts as many comments you added. Makes it much more easier to debug if something fails.

Reply:
Like testKindOf, testTypeOf etc. ?